### PR TITLE
Add support to credential helper

### LINKED ISSUE #250: Add support for credential helper
Would it be possible to add support for the `git credential` helpers?

### PR DESCRIPTION
**Pull request by kassio**
Fixes #250

Related to
- https://github.com/go-git/go-git/issues/28

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -227,28 +229,29 @@ func (c *Config) Validate() error {
...
+	remoteSection     = "remote"
+	submoduleSection  = "submodule"
+	branchSection     = "branch"
+	coreSection       = "core"
+	packSection       = "pack"
+	userSection       = "user"
+	authorSection     = "author"
+	committerSection  = "committer"
+	initSection       = "init"
+	credentialSection = "credential"
```
Comment (kassio) [inline config/config.go:241]: section added

File: plumbing/transport/common.go [hunk 2]
```diff
@@ -166,6 +169,86 @@ func (u *Endpoint) String() string {
...
+func (e Endpoint) credentialQuery() string {
+	return fmt.Sprintf(
+		"protocol=%s\nhost=%s\npath=%s\n\n",
+		e.Protocol,
+		e.Host,
+		e.Path,
+	)
+}
+
+func (e *Endpoint) LoadCredentials(scope config.Scope) error {
```
Comment (kassio) [inline plumbing/transport/common.go:191]: I'm not sure this is the best place to do this.

**Comment by Nintorac** [comment]
This would be very useful, what is the status?

**Comment by kassio** [comment]
Converted this initiative in the #490 issue.

**Pull request closed without merging**